cmd/releasego/update-azure-linux.go

```diff
@@ -0,0 +1,249 @@
+		Name:        "update-azure-linux",
+		Summary:     "Update the Go spec files for Azure Linux.",
+		Description: " ",
+		Handle:      updateAzureLinux,
+	})
+}
+
+func updateAzureLinux(p subcmd.ParseFunc) error {
+	var buildAssetJSON string
+
+	flag.StringVar(&buildAssetJSON, "build-asset-json", "", "[Required] The path of a build asset JSON file describing the Go build to update to.")
+	pat := githubutil.BindPATFlag()
+
+	if err := p(); err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	client, err := githubutil.NewClient(ctx, *pat)
+	if err != nil {
+		return err
+	}
+
+	assets, err := loadBuildAssets(buildAssetJSON)
+	if err != nil {
+		return err
+	}
+
+	golangSpecFileContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", golangSpecFilepath)
+	if err != nil {
+		return err
+	}
+
+	golangSpecFileContent, err = updateSpecFile(assets, golangSpecFileContent)
+	if err != nil {
+		return err
+	}
+
+	prevGoArchiveName, err := extractGoArchiveNameFromSpecFile(golangSpecFileContent)
+	if err != nil {
+		return err
+	}
+
+	// Validation (as described in previous response)
+	if assets.GoSrcURL == "" || assets.GoSrcHash == "" {
+		return fmt.Errorf("invalid or missing GoSrcURL or GoSrcHash in assets.json")
+	}
+
+	golangSignaturesFileContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", golangSignaturesFilepath)
+	if err != nil {
+		return err
+	}
+
+	golangSignaturesFileContent, err = updateSignatureFile(golangSignaturesFileContent, prevGoArchiveName, path.Base(assets.GoSrcURL), assets.GoSrcHash)
+	if err != nil {
+		return err
+	}
+
+	cgManifestContent, err := downloadFileFromRepo(ctx, client, "microsoft", "azurelinux", "3.0-dev", cgManifestFilepath)
+	if err != nil {
+		return err
+	}
+
+	cgManifestContent, err = updateCGManifest(assets, cgManifestContent)
+	if err != nil {
+		return err
+	}
+
+	_ = golangSignaturesFileContent
+	_ = cgManifestContent
+
+	return nil
+}
+
+func loadBuildAssets(assetFilePath string) (*buildassets.BuildAssets, error) {
+	assets := new(buildassets.BuildAssets)
+
+	if err := stringutil.ReadJSONFile(assetFilePath, &assets); err != nil {
+		return nil, fmt.Errorf("error loading build assets: %w", err)
+	}
+
+	return assets, nil
+}
+
+func downloadFileFromRepo(ctx context.Context, client *github.Client, owner, repo, branch, filePath string) ([]byte, error) {
+	fileContent, err := githubutil.DownloadFile(ctx, client, owner, repo, branch, filePath)
+	if err != nil {
+		if errors.Is(err, githubutil.ErrNotExists) {
+			// Handle the specific case of the file not existing
+			return nil, fmt.Errorf("file '%s' not found in repository '%s' on branch '%s'", filePath, repo, branch)
+		} else {
+			// Handle other errors (network issues, authentication, etc.)
+			return nil, fmt.Errorf("failed to download file '%s': %w", filePath, err)
+		}
+	}
+
+	if len(fileContent) == 0 {
+		return nil, fmt.Errorf("downloaded file %s is empty", filePath)
```

Comment:
It seems odd for this func to check this, because it's not necessarily a problem in general cases. I think it would make sense for `updateSignatureFile` etc. to check it.